src/pages/home/ReportScreen.tsx

```diff
@@ -210,6 +210,7 @@ function ReportScreen({
             isOptimisticReport: reportProp?.isOptimisticReport,
             lastMentionedTime: reportProp?.lastMentionedTime,
             avatarUrl: reportProp?.avatarUrl,
+            permissions: reportProp?.permissions,
```

Comment:
When a user signs-in via google after viewing a public room, the `permissions` was not getting updated. A deep comparison was required here to update the dependency. This caused #41955 where the `composer` was not getting displayed even after signing-in via google.